Library/Homebrew/cmd/outdated.rb

```diff
@@ -11,6 +11,10 @@
 #:
 #:    If `--json=`<version> is passed, the output will be in JSON format. The only
 #:    valid version is `v1`.
+#:
+#:    If `--fetch-HEAD` is passed, fetch upstream repository to detect that HEAD
+#:    formula is outdated. Otherwise HEAD-installation is considered outdated if
+#:    new stable or devel version is bumped after that installation.
```

Comment:
> I think we could do it automatically for GitHub, then?

I would avoid it otherwise we may get formula tagged as outdated for a popular repo too often whenever we run `brew upgrade/outdated`. We need a flag to control the behaviors, whether you want prompt update or not.

Besides the time is still relative consuming:
```
$ time brew outdated
brew outdated  0.59s user 0.14s system 79% cpu 0.921 total
$ time brew outdated --fetch-HEAD
neovim/neovim/neovim (HEAD-4dc4efc) < HEAD-abb0928dfbf71d5358b56e888db9ed8ff66d8933
brew outdated --fetch-HEAD  0.54s user 0.10s system 13% cpu 4.918 total
```